ecs-cli/modules/cli/compose/project/project_version.go

```diff
@@ -0,0 +1,43 @@
+package project
+
+import (
+	"io/ioutil"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	yaml "gopkg.in/yaml.v2"
+)
+
+func (p *ecsProject) checkComposeVersion() (string, error) {
+	var composeVersion string
+	for _, file := range p.ecsContext.ComposeFiles {
+		fileVersion, err := getFileVersion(file)
+		if err != nil {
+			return "", err
+		}
+		if len(p.ecsContext.ComposeFiles) > 0 && composeVersion != "" && composeVersion != fileVersion {
+			logrus.Errorf("Compose files must be of the same version. Found: %s and %s", composeVersion, fileVersion)
+		}
+		composeVersion = fileVersion
+	}
+	return composeVersion, nil
+}
+
+func getFileVersion(file string) (string, error) {
+	type ComposeVersion struct {
+		Version string `json:"version"`
+	}
+	version := &ComposeVersion{}
```

Comment:
Hm, seems a little odd to have the type defined in the function itself. Are you thinking this won't be used anywhere else?

Reply:
yes, my thought was this is just an expedient way of extracting the version using yaml.Unmarshal(). Outside of this function the version will be handled as a string or float (pending the discussion above)

Reply:
Yeah, I think the version should still be string regardless of how we parse it. It was just weird to see the type declaration inside the function, but if we're not using it elsewhere it should be fine.